lib/org/chromium/webidl/Canonicalizer.js

```diff
@@ -0,0 +1,129 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'Canonicalizer',
+
+  documentation: `This component expects to receive AST nodes from various
+    sources. Canonical versions of interface definitions will be created
+    from AST nodes of the same source.
+    Please note that this step cannot be distributed. All computations
```

Comment:
More precisely, canonicalization for the same IDL fragment repository cannot be distributed. Distributing between repositories would be a perfectly reasonable thing to do.